### PR TITLE
FUJI-4351 adds modules for job queue and job definition

### PR DESCRIPTION
**Pull request by aeikenberry**
This PR will change this repo from housing only the compute environment into housing the compute environment, batch queue, and batch job definitions as 3 TF modules.

### CODE REVIEW

**Comment by aeikenberry** [comment]
At this point, only Job Queues has been implemented.

File: batch_job_definition/index.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        containerProperties: {
+            environment: [
+                {
+                    name: "ES_ENDPOINT",
+                    value: event.ResourceProperties.EsEndpoint
+                }
+            ],
+            jobRoleArn: event.ResourceProperties.JobRoleArn,
+            image: event.ResourceProperties.ImageName,
+            memory: 50,
```
Comment (mrapptambour) [inline batch_job_definition/index.js]: @aeikenberry param

File: batch_job_definition/index.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            vcpus: 1
```
Comment (mrapptambour) [inline batch_job_definition/index.js]: @aeikenberry param

File: batch_job_definition/index.js (hunk 1 above)
Comment (aeikenberry) [inline batch_job_definition/index.js]: Haven't started that one yet. will definitely.

**Comment by aeikenberry** [comment]
The only thing left for this is figuring out how to pass a map of ENV variables from terraform to CloudFormation to get into the Lambda javascript.

File: batch_compute_environment/main.tf [hunk 3]
```diff
@@ -49,7 +44,7 @@ resource "aws_cloudformation_stack" "batch_compute_environment" {
   name = "${var.ComputeEnvironmentName}-stack"
 
   parameters = {
-    LambdaFunctionName     = "${var.LambdaFunctionName}"
+    LambdaFunctionName     = "fn-${var.ComputeEnvironmentName}-Manager"
```
Comment (mrapptambour) [inline batch_compute_environment/main.tf]: lowercase ;-)

File: batch_job_definition/batch_job_definition.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                  Effect: Allow
+                  Action:
+                    - logs:CreateLogGroup
+                    - logs:CreateLogStream
+                    - logs:PutLogEvents
+                  Resource: "*"
+                -
+                  Effect: Allow
+                  Action:
+                    - iam:PassRole
```
Comment (mrapptambour) [inline batch_job_definition/batch_job_definition.yml]: @aeikenberry got any helpful links for why this line matters? just curious what it does

File: batch_job_definition/main.tf [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+resource "aws_cloudformation_stack" "batch_job_defintion_stack" {
+  name = "${var.JobDefinitionName}-stack"
+
+  parameters = {
+    ImageName                = "${var.ImageName}"
+    JobDefinitionName        = "${var.JobDefinitionName}"
+    BatchJobRoleArn          = "${var.BatchJobRoleArn}"
+    Memory                   = "${var.Memory}"
+    VCpus                    = "${var.VCpus}"
+    JobDefinitionEnvJSON     = "${jsencode("${var.JobDefinitionEnvJSON}")}"
```
Comment (mrapptambour) [inline batch_job_definition/main.tf]: @aeikenberry does it work? Do you love it?

File: batch_job_definition/batch_job_definition.yml (hunk 4 above)
Comment (aeikenberry) [inline batch_job_definition/batch_job_definition.yml]: http://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_passrole.html

**Pull request merged**